src/trace/cursor/cursor_group.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{trace::cursor::Cursor, Timestamp};
+use std::marker::PhantomData;
+
+/// A `CursorGroup` iterates over values associated with a single key of a base
+/// cursor of type `C: Cursor<K, V, R, T>`.
+pub struct CursorGroup<'c, K, T, C> {
```

Comment:
Lots of missing tests here

Reply:
@gz kindly promised to share his proptest-based test harness for cursors, which should improve the situation.